.flake8

```diff
@@ -0,0 +1,14 @@
+[flake8]
+ignore = ANN101, ANN001, ANN002, ANN003, C101, C812, 
+         D100, D101, D102, D103, D104, D105, D106, D107, D203, D400, D401, 
+         E124, E203, I100, I101, I201, I900, S101, S106, SIM105, SIM106, S604, W503, W504, 
```

Comment:
@ungrichtepfl 

- E124, E203 : I wouldn't ignore them
- I100, I101: eventually also them shouldn't be ignore
- I900: this sounds also pretty good to me

I couldn't find an official description of some of the S* rules... from where do these come from?

Reply:
I agree with the second and third point, the code were an artifact from another project.
On the other hand, I would ignore E124 and E203 as those are visual formatting errors, which sometimes disagree with the formatter we use. I would not lint for visual formatting errors in general and use _yapf_ as our "undisputable" ground truth.

Reply:
Flake8 is a plugin based system. Unfortunately this means that the error codes are spread over the following repos (also found in pyproject.toml):
  ```
["flake8-bandit",
  "flake8-bugbear",
  "flake8-builtins",
  "flake8-coding",
  "flake8-commas",
  "flake8-comprehensions",
  "flake8-docstrings",
  "flake8-expression-complexity",
  "flake8-import-order",
  "flake8-isort",
  "flake8-pep3101",
  "flake8-plugin-utils",
  "flake8-polyfill",
  "flake8-pytest",
  "flake8-pytest-style",
  "flake8-requirements",
  "flake8-simplify",
  "flake8-use-fstring",
  "pep8-naming"]
```
I used a selection of the most commonly used plugins. On general python packages are often built with this modular architecture in mind. Makes finding references and docs unfortunately harder.